File: Objet/99-Personne employe/ClassLibraryPersonne/Personne.cs

namespace ClassLibraryPersonne
{
    public abstract class Personne
    {
        private string nom;
        private string prenom;
        private System.DateTime dateDeNaissance;

        public Personne(string _nom, string _prenom, System.DateTime _dateDeNaissance)
        {
            this.Nom = _nom;    
            this.Prenom = _prenom;
            this.DateDeNaissance = _dateDeNaissance;
        }

       
        public string Nom { get => nom; set => nom = value; }
        public string Prenom { get => prenom; set => prenom = value; }
        public DateTime DateDeNaissance { get => dateDeNaissance; set => dateDeNaissance = value; }

        public abstract string Afficher();

        public abstract Personne Evoluer(); 


    }
}

File: PHP/php_objet_exos/personnes/Personne.php
<?php
class Personne
{
    private string $nom;
    private string $prenom;
    private int $age;

    public function __construct(string $nom, string $prenom, string $dateDeNaissance)
    {
        $this->nom = $nom;
        $this->prenom = $prenom;
        $dateDuJour = new DateTime();
        $dateDeNaissance = new DateTime($dateDeNaissance);
        $interval = $dateDuJour->diff($dateDeNaissance);
        $this->age = $interval->y;
    }

    public function getNom()
    {
        return  $this->nom;
    }

    public function getPrenom()
    {
        return  $this->prenom;
    }

    public function getAge()
    {
        return  $this->age;
    }

    public function setNom($nom)
    {
        $this->nom = $nom;
    }
}
